---
title: "Use case 2: Effect of LPS treated microglia EVs in healthy microglia"
author: "Estefania"
date: "2025-03-23"
output: 
  html_document:
    theme: darkly
always_allow_html: true
---

```{r setup, include=FALSE}
r_files <- list.files("../R", pattern = "\\.R$", full.names = TRUE)
lapply(r_files, source)

knitr::opts_chunk$set(echo = TRUE, fig.path='figures/', message=FALSE)
```

This use case explores the effects of EV cargo derived from LPS-treated microglia on healthy microglia.

### 0. Load required packages
```{r 0}

library(Seurat)
library(SeuratObject)
library(tidyverse)
library(dplyr)
library(here)
```

### 1. Load microglia EVs proteomics file
The dataset used in this vignette originates from Santiago et al. (2023), “Identification of State-Specific Proteomic and Transcriptomic Signatures of Microglia-Derived Extracellular Vesicles.” It comprises EVs proteomics data from LPS-stimulated microglia and RNA-seq data from recipient (healthy) microglia exposed to these EVs.

```{r 1}
LPS_microglia_EVs <- read.csv(url("https://zenodo.org/records/17391831/files/LPS_microglia_EVs.csv?download=1"), stringsAsFactors = FALSE)
```

### 2. Extracting the LPS_EV_cargo (more abundant in LPS treated microglia)

```{r 2}
LPS_EV_cargo <- LPS_microglia_EVs %>%
  filter(diff.LPS.Control > 0, Pr..F. < 0.1) %>%
  pull(X)
```

### 3. Importing receiving microglia expression dataset

```{r 3}
receiving_microglia_expression <- readRDS(url("https://zenodo.org/records/17581185/files/receiving_microglia_expression.rds?download=1"))
```

### 4. Defining expressed genes in receiver microglia

```{r 4}
expressed_genes_receiver <- receiving_microglia_expression$UniProt
```

### 5. Loading Nichenet networks from Zenodo

```{r 5}
options(timeout = 600)
organism <- "mouse"

if(organism == "human"){
  lr_network <- readRDS(url("https://zenodo.org/record/7074291/files/lr_network_human_21122021.rds"))
  weighted_networks <- readRDS(url("https://zenodo.org/record/7074291/files/weighted_networks_nsga2r_final.rds?download=1"))
} else if(organism == "mouse"){
  lr_network <- readRDS(url("https://zenodo.org/record/7074291/files/lr_network_mouse_21122021.rds"))
  weighted_networks <- readRDS(url("https://zenodo.org/record/7074291/files/weighted_networks_nsga2r_final_mouse.rds?download=1"))
  
}

lr_network <- lr_network %>% distinct(from, to)
head(lr_network)
```

### 6. Download and load the EV-Net EV_cargo_target_matrix 

Download it from https://zenodo.org/records/15019664/files/EV_cargo_target_matrix.rds and load it into your R environment.

```{r 6}
EV_cargo_target_matrix <- readRDS("~/embo/Netiev/EV_cargo_target_matrix.rds")
#Replace file path with the path in which you stored the EV_cargo_target_matrix rds file
```

### 7. Define expressed interactors and the potential EV cargo

In this step, we identify expressed interactors, which are all proteins expressed in the receiving cell populations, including receptors, downstream signaling proteins, and transcription factors, that could potentially interact with the potential EV cargo (actual EV cargo will be defined in a following step). These expressed interactors will be used to link EV cargo to target genes in downstream analyses.

```{r 7}
all_genes <- unique(rownames(EV_cargo_target_matrix))  
expressed_interactors <- intersect(all_genes, expressed_genes_receiver)

lr_sig <- weighted_networks[["lr_sig"]]
gr <- weighted_networks[["gr"]]

potential_EV_cargo_prot <- lr_sig[lr_sig$to %in% expressed_interactors, "from"]
potential_EV_cargo_tf <- gr[gr$to %in% expressed_interactors, "from"]
potential_EV_cargo <- unique(c(potential_EV_cargo_prot$from, potential_EV_cargo_tf$from))
```

### 8. Reduce the size of the EV_cargo_target_matrix

The EV_cargo_target_matrix is large (3.6 GB). If your system has limited RAM, it is recommended to filter the matrix to keep only the expressed genes in the receiver and the potential EV cargo, which significantly reduces its size. This step is optional for systems with 32 GB of RAM or more.

```{r 8}
EV_cargo_target_matrix <- EV_cargo_target_matrix[rownames(EV_cargo_target_matrix) %in% expressed_genes_receiver, colnames(EV_cargo_target_matrix) %in% potential_EV_cargo]

```

Release memory using the garbage collector

```{r 8.5}
gc()
```

### 9. Define the gene set of interest

To define the gene set of interest, we select the genes with an adjust pvalue `padj < 0.1 ` and an average log2 fold change `(avg_log2FC) ≥ 0.25`.

```{r 9}
geneset_oi <- receiving_microglia_expression %>%
  filter(padj < 0.1 & abs(log2FoldChange) >= 0.25) %>%
  pull(UniProt)
```

```{r 9.5}
geneset_oi <- geneset_oi %>% .[. %in% rownames(EV_cargo_target_matrix)]
```

### 10. Defining background genes

Background genes are all genes expressed in the receiving cell populations. They provide a reference set for statistical analyses and enrichment calculations.

```{r 10}
background_expressed_genes <- expressed_genes_receiver %>% .[. %in% rownames(EV_cargo_target_matrix)]
```

### 11. Get list of EV_cargo

The final set of EV cargo is obtained by selecting proteins that are present in both the actual LPS-derived microglia EV cargo and the list of potential EV cargo (obtained in step 7). This ensures that we focus on proteins that are differentially abundant in the LPS-activated condition and that may influence the expressed genes in the receiving microglia.

```{r 11}
EV_cargo <- intersect(potential_EV_cargo, LPS_EV_cargo)
```

### 12. Perform EV cargo activity analysis

```{r 12}
EV_cargo_activities <- predict_EV_cargo_activities(geneset = geneset_oi,
                                               background_expressed_genes = background_expressed_genes,
                                               EV_cargo_target_matrix = EV_cargo_target_matrix,
                                               potential_EV_cargo = EV_cargo)

EV_cargo_activities <- EV_cargo_activities %>% arrange(-aupr_corrected) %>% mutate(rank = rank(desc(aupr_corrected)))
EV_cargo_activities

```
### 13. Visualization of top-ranked EV cargo

```{r 13}
p_hist_EV_cargo_activity <- ggplot(EV_cargo_activities, aes(x=aupr_corrected)) + 
  geom_histogram(color="black", fill="darkorange")  + 
  geom_vline(aes(xintercept=min(EV_cargo_activities %>% top_n(30, aupr_corrected) %>% pull(aupr_corrected))),
             color="red", linetype="dashed", size=1) + 
  labs(x="EV cargo activity (PCC)", y = "# EV cargo") +
  theme_classic()

p_hist_EV_cargo_activity
```
### 14. We can also visualize the EV cargo activity measure (AUPR) of these top-ranked EV cargo
> **Note:** We have selected the top 30 `best_upstream_EV_cargo` but this number can be changed.

```{r 14}
best_upstream_EV_cargo <- EV_cargo_activities %>% top_n(30, aupr_corrected) %>% arrange(-aupr_corrected) %>% pull(test_EV_cargo)

vis_EV_cargo_aupr <- EV_cargo_activities %>% filter(test_EV_cargo %in% best_upstream_EV_cargo) %>%
  column_to_rownames("test_EV_cargo") %>% select(aupr_corrected) %>% arrange(aupr_corrected) %>% as.matrix(ncol = 1)

(make_heatmap_ggplot(vis_EV_cargo_aupr,
                     "Prioritized EV cargo", "EV cargo activity", 
                     legend_title = "AUPR", color = "darkorange") + 
    theme(axis.text.x.top = element_blank()))  
```

### 15. Infer target genes  of top-ranked EV cargo

```{r 15}
active_EV_cargo_target_links_df <- best_upstream_EV_cargo %>%
  lapply(get_weighted_EV_cargo_target_links,
         geneset = geneset_oi,
         EV_cargo_target_matrix = EV_cargo_target_matrix,
         n = 80) %>%
  bind_rows() %>% drop_na()
```

```{r 15.2}
active_EV_cargo_target_links <- prepare_EV_cargo_target_visualization(
  EV_cargo_target_df = active_EV_cargo_target_links_df,
  EV_cargo_target_matrix = EV_cargo_target_matrix,
  cutoff = 0.5) 
```

```{r 15.5}
order_EV_cargo <- intersect(best_upstream_EV_cargo, colnames(active_EV_cargo_target_links)) %>% rev()
order_targets <- active_EV_cargo_target_links_df$target %>% unique() %>% intersect(rownames(active_EV_cargo_target_links))

vis_EV_cargo_target <- t(active_EV_cargo_target_links[order_targets,order_EV_cargo])

target_genes_heatmap <- make_heatmap_ggplot(vis_EV_cargo_target, "Prioritized EV cargo", "Predicted target genes",
                    color = "purple", legend_title = "Regulatory potential") +
  scale_fill_gradient2(low = "whitesmoke",  high = "purple")
```

```{r 15.7}
target_genes_heatmap
```
### 16. Save plot (optional)
```{r 16}
png("target_genes_heatmap_microglia_EV_cargo_RNAseq.png", res = 300, width = 5000, height = 2000)
print(target_genes_heatmap)
```

### 17. Build an interaction network

Next we will build an interaction network between one of the top-ranked EV cargo: the Mtdh protein, and two of its targets, with biological relevance for inflammation: Ddr1 and Slc7a11 (choice was made after studying the target_genes_heatmap). 

Loading Nichenet's sig_network and gr_network from zenodo

```{r 17}
sig_network <- readRDS(url("https://zenodo.org/records/7074291/files/signaling_network_mouse_21122021.rds"))
gr_network <- readRDS(url("https://zenodo.org/records/7074291/files/gr_network_mouse_21122021.rds"))
```

Inferring EV_cargo-to-target signaling paths

```{r 18}
EV_cargo_oi <- "Mtdh"
targets_oi <- c("Ddr1", "Slc7a11")

active_signaling_network <- get_EV_cargo_signaling_path(EV_cargo_all = EV_cargo_oi,
                                                      targets_all = targets_oi, 
                                                      weighted_networks = weighted_networks,
                                                      EV_cargo_tf_matrix = EV_cargo_target_matrix,
                                                      top_n_regulators = 3,
                                                      minmax_scaling = TRUE) 


graph_min_max <- diagrammer_format_signaling_graph(signaling_graph_list = active_signaling_network,
                                                   EV_cargo_all = EV_cargo_oi, targets_all = targets_oi,
                                                   sig_color = "indianred", gr_color = "steelblue")

DiagrammeR::render_graph(graph_min_max, layout = "tree")

# To export/draw the svg, you need to install DiagrammeRsvg
#graph_svg <- DiagrammeRsvg::export_svg(DiagrammeR::render_graph(graph_min_max, layout = "tree", output = "graph"))
#cowplot::ggdraw() + cowplot::draw_image(charToRaw(graph_svg))
```

Optional: Create a dataframe with annotations of collected data sources supporting the interactions in this network.

```{r 19}
data_source_network <- infer_supporting_datasources(signaling_graph_list = active_signaling_network,
                                                    lr_network = lr_network, sig_network = sig_network, gr_network = gr_network)
head(data_source_network) 
```
